model/lifecycle.go

```diff
@@ -55,25 +55,61 @@ type VersionToRestart struct {
 // SetVersionActivation updates the "active" state of all builds and tasks associated with a
 // version to the given setting. It also updates the task cache for all builds affected.
 func SetVersionActivation(versionId string, active bool, caller string) error {
-	builds, err := build.Find(
-		build.ByVersion(versionId).WithFields(build.IdKey),
-	)
-	if err != nil {
-		return errors.Wrapf(err, "getting builds for version '%s'", versionId)
+	if err := SetVersionActivated(versionId, active); err != nil {
+		return errors.Wrapf(err, "setting activated for version '%s'", versionId)
 	}
-	buildIDs := make([]string, 0, len(builds))
-	for _, build := range builds {
-		buildIDs = append(buildIDs, build.Id)
+	if err := setTaskActivationForVersion(versionId, active, caller); err != nil {
+		return errors.Wrapf(err, "setting activation for tasks in version '%s'", versionId)
 	}
+	return nil
+}
 
-	// Update activation for all builds before updating their tasks so the version won't spend
-	// time in an intermediate state where only some builds are updated
-	if err = build.UpdateActivation(buildIDs, active, caller); err != nil {
-		return errors.Wrapf(err, "setting activation for builds in version '%s'", versionId)
+// setTaskActivationForVersion activates all tasks in a version.
+func setTaskActivationForVersion(versionId string, active bool, caller string) error {
+	q := bson.M{
+		task.VersionKey: versionId,
+		task.StatusKey:  evergreen.TaskUndispatched,
 	}
+	tasksToModify := []task.Task{}
+	// If activating a task, set the ActivatedBy field to be the caller.
+	if active {
+		tasksToModify, err := task.FindAll(db.Query(q).WithFields(task.IdKey, task.DependsOnKey, task.ExecutionKey, task.BuildIdKey))
```

Comment:
You need to use = instead of := to set to tasksToModify, otherwise this variable will only exist within the if statement.
https://go.dev/play/p/1F7QWjlBGDC

Reply:
ah yes i didnt realize what it would do with 2 vars 
defined err outside of the if and fixed